### ISSUE TITLE
Revert changes ignore file semantic

### ISSUE THREAD

**Issue by joaomoreno** [microsoft/vscode-vsce#588]
In https://github.com/microsoft/vscode-vsce/pull/418 we fixed https://github.com/microsoft/vscode-vsce/issues/417 by adopting npm's ignore module for our `.vscodeignore` semantics. This actually broke a lot of extensions:

- https://github.com/microsoft/vscode-vsce/issues/576
- https://github.com/microsoft/vscode-vsce/issues/583
- https://github.com/microsoft/vscode-vsce/issues/584
- https://github.com/microsoft/vscode-vsce/issues/585
- https://github.com/microsoft/vscode-vsce/issues/587

## Semantic differences

It used to be possible to write such a rule:

```
node_modules
!node_modules/highlight.js/styles
```

With npm's `ignore` module, we're force to do this instead:

```
node_modules/*
!node_modules/highlight.js
node_modules/highlight.js/*
!node_modules/highlight.js/styles
```

Apart from being a breaking change, I argue this is a change in behavior for the worse. For those two reasons, I propose to simply revert the change.

- [ ] Revert change
- [ ] Write test cases which lock us in our semantics for future changes
- [ ] Verify this doesn't break extensions which have adopted the new format already

---

Finally, thanks all for the patience and the persistence in filling issues. I apologise for my stubborness. We should've caught this early on and optimized for compatibility. 🙇‍♂️

cc @aeschli @Pustur @fyangstudio @larshp @kitsonk @Timmmm @juli1 @orph @GordonSmith @ntotten @PeterWone

**Comment by GordonSmith** [comment]
@joaomoreno any thoughts on simply supporting the "files" section in package.json to specify which files should go into vsix?

**Comment by joaomoreno** [comment]
No thoughts.

**Comment by larshp** [comment]
note this change was published in 1.95.0

**Comment by Timmmm** [comment]
> @joaomoreno any thoughts on simply supporting the "files" section in package.json to specify which files should go into the vsix?

I agree that would be more intuitive. Definitely feels weird for everything to be published by default!
